lib/utils.pm

```diff
@@ -530,6 +539,13 @@ sub sle_version_at_least {
     die "unsupported SLE $version_variable $version in check";
 }
 
+# Replace X-SPY into X.Y
+sub sle_version_convert {
```

Comment:
it's look like you have more complex scenario in your head , but I suggest move step by step . Currently you declare a function which suppose to get array , but only place where you call it you don't pass even a single param. also I am in doubt how many parameters in same would make sense to by pass here to get something reasonable after "s/-SP/\./gr" filtering. 
At least I would simplify function to one line , at most I would drop it at all

Reply:
And same as for function above everything which inserted into utils.pm suppose to have description